app/controllers/ToDo/UpdateController.js

```diff
@@ -41,35 +39,38 @@ class UpdateController {
      *            ]
      *    }
      */
-    async invoke(request, response, next) {
+    constructor(todoRepository) {
+        this.todoRepository = todoRepository;
+    }
+
+    async invoke(request, response) {
         const todoId = request.params.id;
         const fields = request.body;
 
         fields.creatorId = request.loggedUserId;
+
         if (!fields.userId) {
             fields.userId = request.loggedUserId;
         }
 
-        Todo.findByPk(todoId).then(todo => {
-            if (!todo) {
-                return Todo.create({ ...fields, todoId })
-                    .then(todo => {
-                        return response.status(StatusCodes.CREATED).json(todo);
-                    })
-                    .catch(next);
-            }
+        const todo = await this.todoRepository.findById(todoId);
+
+        if (!todo) {
```

Comment:
if there is no todo then just send 404

Reply:
I think it would be better to improve in next pr - this one + new tests for this and patchController